core.md

```diff
@@ -177,84 +249,105 @@ initialized to 0, fields of type `any` are initialized to `undefined`,
+};
+const RectangleType = new StructType({topLeft: PointType, bottomRight: PointType},
+                                   {defaults: defaults});
+// Instantiate from a source object with one partially and one entirely missing field:
+let rect1 = new RectangleType({topLeft: {x: 10}});
+rect1.topLeft.x === 10;
+rect1.topLeft.y === Number.NEGATIVE_INFINITY;
+rect1.bottomRight.x === Number.POSITIVE_INFINITY;
+rect1.bottomRight.y === Number.POSITIVE_INFINITY;
 ```
 
-The result will be two objects as shown:
+It's an error to provide default values with incorrect types for members with primitive
+type. For members with struct type, only the structure of the default value (and the
+types of any fields that provide default values for members with primitive type contained
+in the struct type) is relevant, not its type.
 
-    line1 ----> +===========+
-                | buffer    | ----> +============+ ArrayBuffer
-                | offset: 0 |       | from: x: 1 |
-                +===========+       |       y: 2 |
-                                    | to:   x: 3 |
-                                    |       y: 4 |
-                                    +============+
+As with builtin default values, overridden default values only apply during
+instantiation, not during assignment. That is, it's still an error to assign an
+incomplete example object to a member that is an embedded struct.
 
-As you can see from the diagram, the typed object `line1` doesn't
-actually store any data itself. Instead, it is simply a pointer into a
-backing store (an array buffer, same as from typed arrays) that
-contains the data itself.
+### Instantiating struct type arrays
 
-*Efficiency note:* The spec has been designed so that, most of the
-time, engines only have to create the backing buffer object and not
-the pointer object `line1`. Instead of creating an object to store the
-buffer and offset, the engine can usually just store the buffer and
-offset directly as synthetic local variables.
+For each type definition, a fixed-sized typed array of instances of that type can be
+created using the type's `.array` constructor.
+
+Just as typed array constructors, typed object array constructors support four
+different overloads:
+
+```js
+const PointType = new StructType({x: float64, y: float64});
+// Creates an instance of length 10, with all entries initialized to default values.
+let points = new PointType.array(10);
+// Creates a copy of `points`.
+let pointsCopy = new PointType.array(points);
+// Creates an instance by iterating over the array-like or iterable source and
+// creating instances of `PointType` for all encountered items.
+let coercedPoints = new PointType.array([new PointType(1, 2), new PointType(10, 20)]);
+// Creates an instance as a view onto the given buffer, starting at the given
+// offset and with the given length, both of which are optional.
+let pointsView = new PointType.array(buffer(points), 3, 3);
```

Comment:
Also, assuming `PointType` were transparent, this also implies that distinct views with the same buffer/offset/length are distinct objects (unlike the views below, which are fat pointers). Is that what we actually want? 

In particular, should the following really be `false`?

    new PointType.array(buffer, 3, 3) === new PointType.array(buffer, 3, 3)

It seems a bit odd for it to be false, given the way that views and identity work below. In particular, imagine that we have an array of `LineType`:

```js
const PointType = new StructType({x: f32, y: f32}, {transparent: true});
const LineType = new StructType({start: PointType, end: PointType}, {transparent: true});
const buffer = ...; // some buffer
const lines1 = new LineType(buffer, x, y);
const lines2 = new LineType(buffer, x, y);
lines1 !== lines2; // true
lines1.start === lines2.start; // true (!)
```

Note the last line, which I think is potentially rather surprising. I believe this would follow, though, from the canonicalization rules, since we have two points at the same offset in the same buffer.

So I think we should break compat (intentionally) with typed arrays here and instead go with `PointType.array.view`.